specification/dartLangSpec.tex

```diff
@@ -3233,39 +3235,40 @@ \subsubsection{The Method \code{noSuchMethod}}
 \end{dartCode}
 
 
-\subsubsection{The Operator `=='}
+\subsubsection{The Operator `==' and Primitive Equality}
 \LMLabel{theOperatorEqualsEquals}
 
 \LMHash{}%
 The operator \lit{==} is used implicitly in certain situations,
 and in particular constant expressions
 (\ref{constants})
-give rise to constraints on that operator.
-In order to specify these constraints just once we introduce the notion of a
-% Neither \syntax nor \lit works, so we fall back to `\code{==}'.
-\IndexCustom{primitive operator `\code{==}'}{%
-  operator `\code{==}'!primitive}:
+give rise to constraints on that operator,
+as well as on the getter \code{hashCode}.
+In order to specify these constraints just once we introduce the notion of
+\Index{primitive equality}:
 
 \begin{itemize}
 \item Every instance of type \code{int} and \code{String}
-  has a primitive operator \lit{==}.
+  has primitive equality.
 \item Every instance of type \code{Symbol}
   which was originally obtained by evaluation of a literal symbol or
   a constant invocation of a constructor of the \code{Symbol} class
-  has a primitive operator \lit{==}.
+  has primitive equality.
 \item Every instance of type \code{Type}
   which was originally obtained by evaluating a constant type literal
   (\ref{dynamicTypeSystem})
-  has a primitive operator \lit{==}.
-\item An instance $o$ has a primitive operator \lit{==}
+  has primitive equality.
+\item An instance $o$ has primitive equality
   if the dynamic type of $o$ is a class $C$,
-  and $C$ has a primitive operator \lit{==}.
-\item The class \code{Object} has a primitive operator \lit{==}.
-\item A class $C$ has a primitive operator \lit{==}
+  and $C$ has primitive equality.
+\item The class \code{Object} has primitive equality.
+\item A class $C$ has primitive equality
   if it does not have an implementation of the operator \lit{==}
+  that overrides the one inherited from \code{Object},
+  and it does not have an implementation of the getter \code{hashCode}
   that overrides the one inherited from \code{Object}.
   \commentary{%
-  In particular, the following have a primitive operator \lit{==}:
+  In particular, the following have primitive equality:
   The null object (\ref{null}),
   function objects obtained by function closurization of
   a static method or a top-level function
```

Comment:
Now we have instantiated tear-off, do we need to mention `possibly instantiated with constant types, if generic"?